lib/puppet/provider/mount.rb

```diff
@@ -26,21 +26,33 @@ def mount
   def remount
     #TRANSLATORS refers to remounting a file system
     info _("Remounting")
-    if resource[:remounts] == :true
+    os = Facter.value(:operatingsystem)
+    supports_remounts = (resource[:remounts] == :true)
+    if supports_remounts && os == 'AIX'
+      remount_with_option("remount")
+    elsif os.match(/^(FreeBSD|DragonFly|OpenBSD)$/)
+      remount_with_option("update")
+    elsif supports_remounts
       mountcmd "-o", "remount", resource[:name]
-    elsif ["FreeBSD", "DragonFly", "OpenBSD"].include?(Facter.value(:operatingsystem))
-      if self.options && !self.options.empty?
-        options = self.options + ",update"
-      else
-        options = "update"
-      end
-      mountcmd "-o", options, resource[:name]
     else
       unmount
       mount
     end
   end
 
+  def remount_with_option(option)
+    if using_explicit_options?
+      options = self.options + "," + option
+    else
+      options = option
+    end
+    mountcmd "-o", options, resource[:name]
+  end
+
+  def using_explicit_options?
+    !self.options.nil? && !self.options.empty?
```

Comment:
I'll give you extra points if you use DeMorgan's law to convert this to `!(self.options.nil? || self.options.empty?)` - that is if you think this is more clear.